Source/Charts/Renderers/LineChartRenderer.swift

```diff
@@ -328,6 +326,9 @@ open class LineChartRenderer: LineRadarRenderer
             
             if j < _xBounds.max
             {
+                // TODO: remove the check.
```

Comment:
This is somewhat incorrect, and the iterator in `XBounds` is theoretically incorrect.
As the point is to iterate through `min + range` - where range *could* go over `max` due to a bouncing animation or something similar.

The thing is that this `XBounds` represents (in all current cases) a range of entries, and going over `max` could mean an `IndexOutOfBounds`.
That's why the check is there.

If the iterator only iterates until `max` - then it's a bug and it's not respecting the animation (range is basically `(max - min) * animation.PhaseX`). 

Also I would change the iterator's `fileprivate init(min: Int, max: Int) {` to `fileprivate init(min: Int, through: Int) {`, as it's really confusing right now, both with the `max` property and with `to vs through` semantics.

Reply:
Hi @danielgindi I came across here while I am working on the 4.0->master merge. 

I'm a little confusing that you saying `the iterator in XBounds is theoretically incorrect.` 

Which iterator you say is wrong? my `return Iterator(min: self.min, max: self.min + self.range)` or `return Iterator(min: self.min, max: self.max)` ? 

because from the old session I wrote here https://github.com/danielgindi/Charts/issues/4093
It seems the iterator of xbounds now is correct:
```swift
    public func makeIterator() -> Iterator {
        return Iterator(min: self.min, max: self.min + self.range)
    }
```
I will rename the iterator to `through` while merging 4.0.